feature-libs/product-configurator/rulebased/components/attribute/types/multi-selection-bundle/configurator-attribute-multi-selection-bundle.component.ts

```diff
@@ -57,13 +58,29 @@ export class ConfiguratorAttributeMultiSelectionBundleComponent
     }
   }
 
+  get withQuantityOnAttributeLevel() {
+    return (
+      this.attribute.dataType ===
+      Configurator.DataType.USER_SELECTION_QTY_ATTRIBUTE_LEVEL
+    );
+  }
+
   get withQuantity() {
     return this.quantityService.withQuantity(
       this.attribute.dataType,
       this.attribute.uiType
     );
   }
 
+  get readOnlyQuantity() {
```

Comment:
for future proofing and error handling, please use `?.` instead of `.` for properties that might be optional. you can confirm which are optional in the model files

Reply:
done